File: week12/ClockAppTemplate/ClockAppTemplate/Clock.cs
using System;
namespace ClockApp
{
    public class Clock
    {
        private Counter _hour;
        private Counter _min;
        private Counter _sec;

        public Clock()
        {
            _hour = new Counter("Hours");
            _min = new Counter("Minutes");
            _sec = new Counter("Seconds");
        }

        public string GetTime()
        {
            string h = _hour.Ticks.ToString("D2");
            string m = _min.Ticks.ToString("D2");
            string s = _sec.Ticks.ToString("D2");
            string time = h + ":" + m + ":" + s;

            return time;
        }

        public void Restart()
        {
            _hour.Reset();
            _min.Reset();
            _sec.Reset();
        }

        public void Tick()
        {
            _sec.Increment();

            if (_sec.Ticks > 60)
            {
                _min.Increment();
                _sec.Reset();

                if (_min.Ticks > 60)
                {
                    _hour.Increment();
                    _min.Reset();
                }
                if (_hour.Ticks > 12)
                {
                    _hour.Reset();
                }
            }


        }
    }
}

File: week12/ClockAppTemplate/ClockAppTemplate/clock.py
from counter import Counter

class Clock:
    def __init__(self):
        self._hours = Counter("Hours")
        self._minutes = Counter("Minutes")
        self._seconds = Counter("Seconds")

    def tick(self):
        self._seconds.increment()
        if self._seconds.ticks == 60:
            self._seconds.reset()
            self._minutes.increment()
            if self._minutes.ticks == 60:
                self._minutes.reset()
                self._hours.increment()
                if self._hours.ticks == 12:
                    self._hours.reset()

    def get_time(self) -> str:
        # The :02 ensures two digits with leading zero if needed
        return f"{self._hours.ticks:02}:{self._minutes.ticks:02}:{self._seconds.ticks:02}"

    def reset(self):
        self._hours.reset()
        self._minutes.reset()
        self._seconds.reset()